### PR TITLE
chore: update go-libp2p to v0.27.3

### PR DESCRIPTION
**Pull request by marten-seemann**
## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: cli/net.go [hunk 1]
```diff
@@ -445,9 +445,6 @@ var NetReachability = &cli.Command{
 		}
 
 		fmt.Println("AutoNAT status: ", i.Reachability.String())
-		if i.PublicAddr != "" {
-			fmt.Println("Public address: ", i.PublicAddr)
-		}
```
Comment (marten-seemann) [inline cli/net.go:450]: I removed this log (?) output. It was misleading to begin with, a libp2p node doesn't only have a single public address. The correct way to obtain the addresses is to use the `Addrs` method on the host.

File: node/impl/net/net.go [hunk 2]
```diff
@@ -143,19 +143,7 @@ func (a *NetAPI) NetAutoNatStatus(ctx context.Context) (i api.NatInfo, err error
 		}, nil
 	}
 
-	var maddr string
-	if autonat.Status() == network.ReachabilityPublic {
-		pa, err := autonat.PublicAddr()
```
Comment (marten-seemann) [inline node/impl/net/net.go:148]: We removed the `PublicAddr` method from AutoNAT.

File: api/api_net.go [hunk 3]
```diff
@@ -73,5 +73,4 @@ type CommonNet interface {
 
 type NatInfo struct {
 	Reachability network.Reachability
-	PublicAddr   string
```
Comment (magik6k) [inline api/api_net.go:76]: Can we change this to an array, and populate this from Addrs? We have a bunch of tools / docs which want this API / CLI (mostly there some docs that refer to the Public address output from cli)

**Comment by jacobheun** [comment]
What's the state of this? I was hoping to get go-fil-markets updated but it needs this update for libp2p.

**Review by magik6k — APPROVED** [review]

**Pull request merged**